### ISSUE TITLE
Update AST to hold const generics

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/syn#227]
The Generics struct now needs to hold `lifetimes`, `ty_params`, and **`const_params`**.

https://github.com/rust-lang/rust/issues/44580

```rust
struct RectangularArray<T, const WIDTH: usize, const HEIGHT: usize> {
    array: [[T; WIDTH]; HEIGHT],
}
```

**Comment by eddyb** [comment]
That's not how we're doing it in the compiler, see https://github.com/rust-lang/rust/pull/45930.

**Comment by dtolnay** [comment]
Implemented in #261.
